compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -76,6 +76,7 @@ pub(crate) fn eval_nullary_intrinsic<'tcx>(
             ty::Adt(ref adt, _) => {
                 ConstValue::from_machine_usize(adt.variants().len() as u64, &tcx)
             }
+            ty::TyAlias(def_id, substs) => eval_nullary_intrinsic(tcx, param_env, *def_id, substs)?,
```

Comment:
The `def_id` passed to `eval_nullary_intrinsic` is the def_id of the callee, i.e., it determines the intrinsic that is being called. How does it make sense to pass it the def_id from `TyAlias`...?

Reply:
I expected it to be normalized inside `eval_nullary_intrinsic`. I guess I was wrong.